app/controllers/devise/two_factor_authentication_controller.rb

```diff
@@ -123,17 +132,9 @@ def handle_second_factor_locked_resource
       sign_out
     end
 
-    def prompt_for_otp_reentry
-      if current_otp_method == :totp
-        show_totp_prompt
-      else
-        show_direct_otp_prompt
-      end
-    end
-
     def send_user_otp
       current_user.send_new_otp(otp_method: current_otp_method)
-      show_direct_otp_prompt
+      redirect_to otp_confirm_path(otp_method: current_otp_method)
```

Comment:
Actually, I take that back. The conditionals in the confirm method were added in this PR. Since the first conditional is not needed, moving rescue code verification to its own controller should take care of this by restoring the confirm method to its previous state.